rosidl_generator_cpp/test/test_interfaces.cpp

```diff
@@ -106,3 +113,186 @@ TEST(Test_rosidl_generator_traits, has_fixed_size) {
   static_assert(array_primitives_size == 3 * primitive_size, "Wrong size");
   static_assert(primitive_static_array_size == 3 * primitive_size, "Wrong size");
 }
+
+TEST(Test_messages, primitives_static) {
+  rosidl_generator_cpp::msg::PrimitivesStatic message;
+
+  message.bool_value = false;
+  ASSERT_EQ(false, message.bool_value);
+  message.bool_value = true;
+  ASSERT_EQ(true, message.bool_value);
+
+  message.byte_value = 0;
+  ASSERT_EQ(0, message.byte_value);
+  message.byte_value = 255;
+  ASSERT_EQ(255, message.byte_value);
+
+  message.char_value = CHAR_MIN;
+  ASSERT_EQ(CHAR_MIN, message.char_value);
+  message.char_value = CHAR_MAX;
+  ASSERT_EQ(CHAR_MAX, message.char_value);
+
+  message.float32_value = FLT_MIN;
+  ASSERT_EQ(FLT_MIN, message.float32_value);
+  message.float32_value = FLT_MAX;
+  ASSERT_EQ(FLT_MAX, message.float32_value);
+
+  message.float64_value = DBL_MIN;
+  ASSERT_EQ(DBL_MIN, message.float64_value);
+  message.float64_value = DBL_MAX;
+  ASSERT_EQ(DBL_MAX, message.float64_value);
+
+  message.int8_value = INT8_MIN;
+  ASSERT_EQ(INT8_MIN, message.int8_value);
+  message.int8_value = INT8_MAX;
+  ASSERT_EQ(INT8_MAX, message.int8_value);
+
+  message.uint8_value = 0;
+  ASSERT_EQ(0, message.uint8_value);
+  message.uint8_value = UINT8_MAX;
+  ASSERT_EQ(UINT8_MAX, message.uint8_value);
+
+  message.int16_value = INT16_MIN;
+  ASSERT_EQ(INT16_MIN, message.int16_value);
+  message.int16_value = INT16_MAX;
+  ASSERT_EQ(INT16_MAX, message.int16_value);
+
+  message.uint16_value = 0;
+  ASSERT_EQ(0, message.uint16_value);
+  message.uint16_value = UINT16_MAX;
+  ASSERT_EQ(UINT16_MAX, message.uint16_value);
+
+  message.int32_value = INT32_MIN;
+  ASSERT_EQ(INT32_MIN, message.int32_value);
+  message.int32_value = INT32_MAX;
+  ASSERT_EQ(INT32_MAX, message.int32_value);
+
+  message.uint32_value = 0;
+  ASSERT_EQ(0, message.uint32_value);
+  message.uint32_value = UINT32_MAX;
+  ASSERT_EQ(UINT32_MAX, message.uint32_value);
+
+  message.int64_value = INT64_MIN;
+  ASSERT_EQ(INT64_MIN, message.int64_value);
+  message.int64_value = INT64_MAX;
+  ASSERT_EQ(INT64_MAX, message.int64_value);
+
+  message.uint64_value = 0;
+  ASSERT_EQ(0, message.uint64_value);
+  message.uint64_value = UINT64_MAX;
+  ASSERT_EQ(UINT64_MAX, message.uint64_value);
+}
+
+/**
+ * Check if the arrays are identical element by element.
+ */
+template<class T, std::size_t SIZE>
+void assert_array(const std::array<T, SIZE> & a, const std::array<T, SIZE> & b)
+{
+  int i;
+
+  for (i = 0; i < SIZE; i++) {
+    ASSERT_EQ(a[i], b[i]);
+  }
+}
```

Comment:
Why is this function necessary? There should be a comparison operator for `std::array` which does that already.